src/Features/ERDDAP/Map/WLPlatformLayer.tsx

```diff
@@ -74,29 +75,52 @@ export const WLPlatformLayer = ({ platform, selected, old = false }: PlatformLay
   const path = usePathname()
   const searchParams = useSearchParams()
   const [floodThreshold, setFloodThreshold] = useState<string>("")
+  const [predictedFloodThreshold, setPredictedFloodThreshold] = useState<string>("")
   const [display, setDisplay] = useState("grey")
+  const [predictedDisplay, setPredictedDisplay] = useState("gray")
   let radius: number
   if (selected) {
-    radius = window.innerWidth > adjustPxWidth ? 10 : 15
+    radius = window.innerWidth > adjustPxWidth ? 16 : 21
   } else {
-    radius = window.innerWidth > adjustPxWidth ? 5 : 10
+    radius = window.innerWidth > adjustPxWidth ? 10 : 15
   }
   const isSensorPage = path.includes("sensor")
 
+  const getObservedWaterDisplay = (currentWaterLevel) => {
+    const value = currentWaterLevel?.value
+    const waterLevelThresholds = getWaterLevelThresholdsMapRawComp(currentWaterLevel?.flood_levels)
+    const surpassedThreshold = getSurpassedThreshold(value, waterLevelThresholds)
+    setFloodThreshold(surpassedThreshold)
+    const opacity = selected ? "e6" : "a0"
+    const display = floodLevelThresholdColors(surpassedThreshold, old, opacity, platform)
+    setDisplay(display)
+  }
+
+  const getPredictedWaterDisplay = (predictedWaterLevel) => {
+    const highValue = predictedWaterLevel.extrema_values.max?.value
+    const predWaterLevelThresholds = getWaterLevelThresholdsMapRawComp(predictedWaterLevel?.flood_levels)
+    const surpassedThreshold = getSurpassedThreshold(highValue, predWaterLevelThresholds)
+    setPredictedFloodThreshold(surpassedThreshold)
+    const opacity = selected ? "e6" : "a0"
+    const display = floodLevelThresholdColors(surpassedThreshold, old, opacity, platform)
+    setPredictedDisplay(display)
+  }
   useEffect(() => {
     const currentWaterLevel = platform.properties.readings.find(
       (r) => r.flood_levels.length && r.variable !== "predictedWL",
```

Comment:
We'll probably want to key this off of `r.type` and r.data_type.standard_name` rather than a specific variable name, since that can be different per source dataset.

Reply:
Great. Thanks for this--It was something I thought might be variable.